krabs/krabs/ut.hpp

```diff
@@ -238,12 +238,14 @@ namespace krabs { namespace details {
         // for MOF providers, EventHeader.Provider is the *Message* GUID
         // we need to ask TDH for event information in order to determine the
         // correct provider to pass this event to
-        auto schema = get_event_schema_from_tdh(record);
-        auto eventInfo = reinterpret_cast<PTRACE_EVENT_INFO>(schema.get());
-        for (auto& provider : trace.providers_) {
-            if (eventInfo->ProviderGuid == provider.get().guid_) {
-                provider.get().on_event(record, trace.context_);
-                return;
+        TDHSTATUS status = ERROR_SUCCESS;
```

Comment:
Before this may have thrown but now it will not. That seems like a big difference in behavior. Need to understand if anything downstream expected this to be throwing. That said, it's _probably_ better this way, but I wonder if there are any cases for TDHSTATUS where it would make the most sense to fail here.

Reply:
It seems that no one can properly catch the exceptions thrown in forward_events, which lead to an interruption in ProcessTrace. If someone needs to catch exceptions out of ProcessTrace, then there indeed will be an impact on compatibility. However, overall I feel that the behavior after PR is more in line with subjective expectations. If necessary, is it required to add a bool flag to ensure compatibility?